components/helpers/applyFilter.ts

```diff
@@ -0,0 +1,139 @@
+/**
+ * @description Sorts an array of objects based on a string property called 'value'.
+ * @param arr Array of objects with a 'value' property of type string.
+ * @returns Sorted array of objects.
+ */
+export function sortFilter(arr: { value: string }[]): { value: string }[] {
+  return arr.sort((a, b) => {
+    if (a.value < b.value) {
+      return -1;
+    }
+    if (a.value > b.value) {
+      return 1;
+    }
+
+    return 0;
+  });
+}
+
+/**
+ * @description Apply filters to data and update the filters.
+ * @param checks Array of filter criteria objects.
+ * @param data Array of data objects to filter.
+ * @param setFilters Function to update the filters.
+ */
+export const applyFilterList = (checks: { name: string }[],
+  data: { [key: string]: any }[],
+  setFilters: (lists: { [key: string]: { value: string; text: string }[] }) => void): void => {
```

Comment:
kindly move such (complex object types) types to top of this file instead of inlining them.